app/rfid.py

```diff
@@ -11,21 +17,17 @@ def __init__(self):
             parity=serial.PARITY_NONE,
             stopbits=serial.STOPBITS_ONE,
             bytesize=serial.EIGHTBITS,
-            timeout=0,
+            timeout=10
         )
-        # ser.write(b'SD2\r\n')
+        # ser.write(b'SD2\r\n') # Sets transponder default to FDX-B(animal tag)
 
     def read(self):
-        # ser.reset_input_buffer()
+        ser.reset_input_buffer()
         ser.reset_output_buffer()
-        ser.write(b'rat\r\n')
-        a = ser.read(34).decode('utf-8')
-        if '?1' in a:
+        ser.write(b'sd2\r\n')
+        a = ser.read_until(size=19).decode('utf-8')
+        print(a)
+        if len(a) < 16:
             return 'TagNotPresent'
-
         else:
-            a = str(a)  # change to a proper string
-            a = a[0:16]  # just send unique data part withou error code
-            ser.reset_input_buffer()
-            ser.reset_output_buffer()
             return a
```

Comment:
Where do you ensure you have only 16 bytes not 19.  I think the slicing should be in the rfid class.